packages/core/src/Constants/form-error-messages.js

```diff
@@ -22,15 +22,16 @@ export const FORM_ERROR_MESSAGES = {
     decimalPlaces: decimals =>
         localize('Up to {{decimal_count}} decimal places are allowed.', { decimal_count: decimals }),
     value: value => localize('Should be {{value}}', { value }),
-    betweenMinMax: (min_value, max_value) =>
+    betweenMinMax: (min_value, max_value, decimals) =>
         localize('Should be between {{min_value}} and {{max_value}}', {
-            min_value,
-            max_value,
+            min_value: Number(min_value).toFixed(decimals || 0),
+            max_value: Number(max_value).toFixed(decimals || 0),
         }),
```

Comment:
Refactor `min_value` and `max_value` properties in order to display big numbers without scientific notation (e.g instead `4.2e-7` we should show `0.00000042`).